pkg/backends/utils.go

```diff
@@ -57,11 +54,11 @@ func dialerSession(ctx context.Context, redial bool, redialDelay time.Duration,
 				}
 				select {
 				case <-time.After(redialDelayInc.Duration):
+					redialDelayInc.NextDelay()
```

Comment:
How about having the IncrementalDuration object have a method that returns a channel and increments the delay internally?  That way you can just do
```
case <-redialDelayInc.NextTimeout():
```
which might be implemented as something like:
```
func (ID *IncrementalDuration) NextTimeout() chan struct{} {
	ch := time.After(ID.Duration)
	ID.Duration = time.Duration(math.Min(ID.Multiplier*float64(ID.Duration), float64(ID.MaxDuration)))
	return ch
}
```
And maybe the struct values like `Duration` ought to be private?  I'm not sure what value there is for letting user code mess with our internal variables.

Reply:
okay I've added a `NextTimeout()` that returns a `time.After` type of channel. To aid with the unit test, I have a method to return current duration, and a method to increase the duration -- but these aren't intended to be used directly.